If the feature is in there before you start liberty:dev exits:
```
[INFO] Running liberty:generate-features
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.090 s
[INFO] Finished at: 2026-05-08T16:25:15-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.12.1-SNAPSHOT:dev (default-cli) on project demo-devmode-maven: If you would like to continue using auto-generation of features, please remove all versionless features from your server configuration. If you would like to continue using versionless features, you can turn off auto-generation of features by using the g hot key to toggle off auto-generation of features. To disable the automatic generation of features, type 'g' and press 'Enter' once dev mode is running or restart dev mode with -DgenerateFeatures=false. -> [Help 1]
```
If you modify server.xml after liberty:dev starts you see 
```
[INFO] Copied file: /Users/paulg/WASDevEx/testing2026/demo-devmode-jakarta/src/main/liberty/config2/server.xml to: /Users/paulg/WASDevEx/testing2026/demo-devmode-jakarta/target/.libertyFeatures/server.xml
[INFO] Running liberty:generate-features
[ERROR] If you would like to continue using auto-generation of features, please remove all versionless features from your server configuration. If you would like to continue using versionless features, you can turn off auto-generation of features by using the g hot key to toggle off auto-generation of features.
To disable the automatic generation of features, type 'g' and press Enter.
[INFO] Copied file: /Users/paulg/WASDevEx/testing2026/demo-devmode-jakarta/src/main/liberty/config2/server.xml to: /Users/paulg/WASDevEx/testing2026/demo-devmode-jakarta/target/liberty/wlp/usr/servers/defaultServer/server.xml
```
If you call generate-features from the command line:
```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.841 s
[INFO] Finished at: 2026-05-08T16:29:17-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.12.1-SNAPSHOT:generate-features (default-cli) on project demo-devmode-maven: Versionless features are detected in the server configuration. If you would like to use auto-generation of features, remove all versionless features from your server configuration. If you would like to continue using versionless features, you cannot use the generate-features mojo. -> [Help 1]
```

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java
@@ -64,10 +64,9 @@ public class GenerateFeaturesMojo extends PluginConfigSupport {
     public static final String NO_NEW_FEATURES_COMMENT = "No additional features generated";
     public static final String NO_CLASSES_DIR_WARNING = "Could not find classes directory to generate features against. Liberty features will not be generated. "
             + "Ensure your project has first been compiled.";
-    public static final String VERSIONLESS_FEATURE_DETECTED_DEVMODE = "If you would like to continue using auto-generation of features, "
-            + "please remove all versionless features from your server configuration. "
-            + "If you would like to continue using versionless features, you can use the 'g' hot key "
-            + "to toggle off auto-generation of features.";
+    public static final String VERSIONLESS_FEATURE_DETECTED_DEVMODE = "Versionless features are detected in the server configuration. "
+            + "If you would like to continue using the automatic generation of features, remove all versionless features from your server configuration. "
+            + "If you would like to continue using versionless features, you can disable the automatic generation of features.";
     public static final String VERSIONLESS_FEATURE_DETECTED = "Versionless features are detected in the server configuration. "
             + "To use the 'generate-features' goal, remove all versionless features from your server configuration. "
             + "If you would like to continue using versionless features, you cannot use the 'generate-features' goal.";

